File: Digital Business Enablement/buy.it-api/src/main/java/br/com/fiap/buy/it/model/Status.java
package br.com.fiap.buy.it.model;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.*;

@Getter
@Setter
@Data
@NoArgsConstructor
@AllArgsConstructor
@ToString
@Entity
@Table(name = "STATUS", uniqueConstraints = {
        @UniqueConstraint(name = "UK_NOME_STATUS", columnNames = "NOME_STATUS")
})
public class Status {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "SQ_STATUS")
    @SequenceGenerator(name = "SQ_STATUS", sequenceName = "SQ_STATUS", allocationSize = 1)
    @Column(name = "ID_STATUS")
    private Long id;

    @Column(name = "NOME_STATUS", nullable = false)
    @NotBlank(message = "O nome do status não pode estar vazio.")
    private String nome;

}

File: Enterprise Application Development/buyit/Models/Status.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace br.com.fiap.buy.it.model
{
    [Table("STATUS")]
    public class Status
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Sequence)]
        [Column("ID_STATUS")]
        public long Id { get; set; }

        [Required(ErrorMessage = "O campo nome não pode estar vazio.")]
        [Column("NOME_STATUS")]
        public string Nome { get; set; }
    }
}